PCHR-2848: Updates to "My Details" Block

mickadoo opened the pull request:
## Overview

The "My Details" block has a few different displays. We're using different ones on the `/hr-details` page and `/dashboard`. These will be unified and old displays removed.

## Before

- `/hr-details` used a different display for the "My Details" view than on `/dashboard`.
- The master display for this view did not show Contact ID and Birth Date
- The address in the "My Address View" showed each address field separately
- The "My Details" view had 4 displays

#### master Display

![image](https://user-images.githubusercontent.com/6374064/32268761-340d07e0-bee8-11e7-943a-c693b508794e.png)

#### hr-details Display

![image](https://user-images.githubusercontent.com/6374064/32237522-0e8776e4-be5d-11e7-9411-dd9f9bf98c2a.png)

## After

- `/hr-details` uses the same "My Details" (master) display as on `/dashboard`.
- The master display for this view adds Contact ID and Birth Date
- The address in the "Contact Information (renamed)" is aggregated to a single field
- 2 obsolete displays are removed from "My Details" 

#### both hr-details and master

![image](https://user-images.githubusercontent.com/6374064/32237678-7f9d5006-be5d-11e7-998c-49bcf342c8a3.png)

## Technical Details

Previously we were adding all the address fields to the view, hiding them, and then using a global text field and replacement tokens to show the address as one field. It seems cleaner to use a custom handler so I added one.

---

- [x] Tests Pass

tunbola commented on civihr_employee_portal/views/includes/civihr_employee_portal_handler_aggregated_address.inc:
```diff
@@ -0,0 +1,81 @@
+   * @return string
+   *   The formatted address
+   */
+  private function formatAddress($address) {
```
Can you add a small description to the function.

tunbola commented on civihr_employee_portal/views/includes/civihr_employee_portal_handler_aggregated_address.inc:
```diff
@@ -0,0 +1,81 @@
+   *   The formatted address
+   */
+  private function formatAddress($address) {
+    $format = '%s<br/>%s<br/>%s %s<br/>%s<br />%s';
```
Seems the space before the closing tag for the last `<br />` is not consistent with the others.
probably make all to have a space before the closing tag.

tunbola commented on civihr_employee_portal/views/includes/civihr_employee_portal_handler_aggregated_address.inc:
```diff
@@ -0,0 +1,81 @@
+  private function formatAddress($address) {
+    $format = '%s<br/>%s<br/>%s %s<br/>%s<br />%s';
+
+    return sprintf(
```
Good use of `sprintf` here.

tunbola commented:
@mickadoo , The before screenshot still shows contact ID and Birth date but the notes say otherwise. Is it the correct one?

mickadoo commented:
@tunbola 

> The before screenshot still shows contact ID and Birth date but the notes say otherwise. Is it the correct one?

Sorry, it was a bit confusing. Before the change the "master" display did not show contact ID and birth date. After the change it does, and now both `/dashboard` and `/hr-details` use the master display. 

I've updated the description and added another screenshot of the previous master display.
